builtins/amp-img.js

```diff
@@ -64,17 +64,12 @@ export class AmpImg extends BaseElement {
   }
 
   /** @override */
-  prerenderAllowed() {
-    return true;
-  }
-
-  /** @override */
   isRelayoutNeeded() {
     return true;
   }
 
   /** @override */
-  layoutCallback() {
+  prerenderCallback() {
```

Comment:
The difference with current state, `prerenderAllowed` really says "do layout on prerender". Given this, runtime actually tracks this and correctly assigns "element is laid out" state, not just prerendered.

Otherwise, there's no problem for image case and I might be overthinking this.